kotlinx-coroutines-core/jvm/src/scheduling/CoroutineScheduler.kt

```diff
@@ -658,11 +657,21 @@ internal class CoroutineScheduler(
         var nextParkedWorker: Any? = NOT_IN_STACK
 
         /*
-         * The delay until at least one task in other worker queues will  become stealable.
+         * The delay until at least one task in other worker queues will become stealable.
          */
         private var minDelayUntilStealableTaskNs = 0L
 
-        private var rngState = Random.nextInt()
+        /**
+         * The state of embedded Marsaglia xorshift random number generator, used for work-stealing purposes.
+         * It is initialized with a seed.
+         */
+        private var rngState: Int = run {
+            // This could've been Random.nextInt(), but we are shaving an extra initialization cost, see #4051
+            val seed = System.nanoTime().toInt()
+            // rngState shouldn't be zero, as required for the xorshift algorithm
+            if (seed != 0) return@run seed
+            42
```

Comment:
It's more conventional to return `4`: https://xkcd.com/221/ `42` is the answer to a question, but `rngState` is not giving us an answer of any kind, so this constant is not really applicable.